src/zdaemon/tests/tests.py

```diff
@@ -79,7 +79,7 @@ def dont_hang_when_program_doesnt_start():
     >>> write('conf',
     ... '''
     ... <runner>
-    ...   program sleep
+    ...   program sleep .1
```

Comment:
AFAICT this test is testing a workaround for a bug that isn't present on Python 3.5.  So maybe just skip it if sys.version_info >= (3, 5)?

OTOH if the bug comes back, we want a test to make sure the workaround works.

What can happen if you try to start a program that fails?

1. zdaemon starts up fine and silently and repeatedly tries to start the program in the background (this is happening on Python 3.5)
2. zdaemon fails to start up (this is happening on Python <= 3.4)
3. zdaemon hangs forever (this is what was happening on Python <= 3.5 before the workaround that this test is trying to test)

Either 1 or 2 is acceptable (although I prefer 2, since that way the admin knows at once that something went wrong).  This test tries to enforce 2.